catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/MigrationFileWriter.java

```diff
@@ -0,0 +1,101 @@
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.codice.ddf.migration.ExportMigrationException;
+import org.codice.ddf.migration.MigrationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.MetacardImpl;
+
+/**
+ * Responsible for all low-level IO operations pertaining to migration.
+ */
+public class MigrationFileWriter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MigrationFileWriter.class);
+
+    /**
+     * Creates the directory for the catalog export if it doesn't exist.
+     * <p>
+     * If the directory does exist, work based on the assumption it was cleaned by the parent
+     * process in the management logic of inport and export. There is no point to delegate
+     * low level directory operations to each and every migratable when they can be abstracted
+     * into the base logic.
+     *
+     * @param exportPath The path representing the directory to create.
+     * @throws MigrationException thrown if the export directory doesn't exist and could not be created
+     */
+    public void createExportDirectory(Path exportPath) throws MigrationException {
+        try {
+            final File exportDir = exportPath.toFile();
+            FileUtils.forceMkdir(exportDir);
+        } catch (IOException e) {
+            LOGGER.error("IO Exception during FileUtils.forceMkdir", e.getMessage(), e);
+            throw new ExportMigrationException(e.getMessage());
+        }
+    }
+
+    /**
+     * Writes the metacards in the provided results list to the provided file.
+     *
+     * @param exportFile The file to write to.
+     * @param results    The metacards to write into the file.
+     */
+    public void writeMetacards(File exportFile, final List<Result> results) throws IOException {
+        if (!exportFile.exists()) {
+            boolean success = exportFile.createNewFile();
+            if (!success) {
+                LOGGER.warn("Was not able to create new file for metacard export");
+            }
+        }
+
+        try (
+                FileOutputStream fileStream = createFileStream(exportFile);
+                BufferedOutputStream bufferedStream = createBufferedStream(fileStream);
+                ObjectOutputStream objectStream = createObjectStream(bufferedStream);
+        ) {
+            for (final Result result : results) {
+                Metacard metacard = new MetacardImpl(result.getMetacard());
+                objectStream.writeObject(metacard);
+            }
+            bufferedStream.flush();
+        }
+    }
+
+    ObjectOutputStream createObjectStream(OutputStream source) throws IOException {
```

Comment:
I don't like this at all. We don't do this in other classes to test them. Why not just actually read/write files from/to the Junit temp directory and use files in the test resources? That is what we normally do for situations like this.